pkg/backup/client.go

```diff
@@ -797,37 +805,54 @@ func SendBackup(
 	client kvproto.BackupClient,
 	req kvproto.BackupRequest,
 	respFn func(*kvproto.BackupResponse) error,
+	resetFn func() (kvproto.BackupClient, error),
 ) error {
-	log.Info("try backup",
-		zap.Stringer("StartKey", utils.WrapKey(req.StartKey)),
-		zap.Stringer("EndKey", utils.WrapKey(req.EndKey)),
-		zap.Uint64("storeID", storeID),
-	)
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
-	bcli, err := client.Backup(ctx, &req)
-	if err != nil {
-		log.Error("fail to backup", zap.Uint64("StoreID", storeID))
-		return err
-	}
-	for {
-		resp, err := bcli.Recv()
-		if err != nil {
-			if err == io.EOF {
-				log.Info("backup streaming finish",
-					zap.Uint64("StoreID", storeID))
-				break
-			}
-			return errors.Annotatef(err, "Store: %d close the connection", storeID)
-		}
-		// TODO: handle errors in the resp.
-		log.Info("range backuped",
-			zap.Stringer("StartKey", utils.WrapKey(resp.GetStartKey())),
-			zap.Stringer("EndKey", utils.WrapKey(resp.GetEndKey())))
-		err = respFn(resp)
+backupLoop:
+	for retry := 0; retry < backupRetryTimes; retry++ {
+		log.Info("try backup",
+			zap.Stringer("StartKey", utils.WrapKey(req.StartKey)),
+			zap.Stringer("EndKey", utils.WrapKey(req.EndKey)),
+			zap.Uint64("storeID", storeID),
+		)
+		bcli, err := client.Backup(ctx, &req)
```

Comment:
Is it possible to test this by injecting a failpoint here🤔?

Reply:
already add a test, PTAL